src/validation/__tests__/OverlappingFieldsCanBeMerged-test.js

```diff
@@ -197,6 +197,34 @@ describe('Validate: Overlapping fields can be merged', () => {
     );
   });
 
+  it('allows different args where no conflict is possible on fragments', () => {
+    // This is valid since no object can be both a "Dog" and a "Cat", thus
+    // these fields can never overlap.
+    expectPassesRule(
+      OverlappingFieldsCanBeMerged,
+      `
+        {
+        pet {
+          ... on Dog {
+            ...X
+          }
+          ... on Cat {
+            ...Y
+          }
+        }
+      }
+
+      fragment X on Pet {
+        name(surname: true)
```

Comment:
This example currently fails on the Facebook GraphQL server (which does not depend on graphql-js for validation).

I would need to think about this example and whether it can be valid. I'm not sure whether it's a bug in both `graphql-js` and the Facebook server, or if it's intended behavior.